src/query/sql/src/planner/optimizer/heuristic/prune_unused_columns.rs

```diff
@@ -278,7 +279,28 @@ impl UnusedColumnPruner {
                 ))
             }
 
-            RelOperator::DummyTableScan(_) => Ok(expr.clone()),
+            RelOperator::MaterializedCte(cte) => {
+                let left_output_column = RelExpr::with_s_expr(expr)
+                    .derive_relational_prop_child(0)?
+                    .output_columns
+                    .clone();
+                let right_used_column = RelExpr::with_s_expr(expr)
+                    .derive_relational_prop_child(1)?
+                    .used_columns
+                    .clone();
+                // Get the intersection of `left_used_column` and `right_used_column`
+                let left_required = left_output_column
+                    .intersection(&right_used_column)
```

Comment:
Why intersect left_output_column and right_used_column.

Reply:
left side only needs to keep the columns that are used by right side.